File: js/programmers/level.1/prgm_42862_1.js
function solution(n, lost, reserve) {
  let lostNum = lost.filter((v) => !reserve.includes(v)).sort((a, b) => a - b);
  let reserveNum = reserve
    .filter((v) => !lost.includes(v))
    .sort((a, b) => a - b);

  reserveNum.forEach((base) => {
    if (lostNum.includes(base - 1)) {
      lostNum = lostNum.filter((v) => v !== base - 1);
    } else if (lostNum.includes(base + 1)) {
      lostNum = lostNum.filter((v) => v !== base + 1);
    }
  });

  return n - lostNum.length;
}

File: python/programmers/level.1/prgm_42862_1.py
def solution(n, lost, reserve):
    lost_list = sorted([num for num in lost if num not in reserve])
    reserve_list = sorted([num for num in reserve if num not in lost])

    for num in reserve_list:
        if num - 1 in lost_list:
            lost_list.remove(num - 1)
        elif num + 1 in lost_list:
            lost_list.remove(num + 1)
    
    return n - len(lost_list)